xapian-core/tests/api_cluster.cc

```diff
@@ -80,3 +105,35 @@ DEFINE_TESTCASE(round_robin1, backend)
     }
     return true;
 }
+
+DEFINE_TESTCASE(stem_stopper1, backend)
+{
+    Xapian::Stem stemmer("english");
+    // By default, stemming strategy used is STEM_SOME
+    Xapian::StemStopper stopper(stemmer);
+    std::string term = "the";
+    stopper.add(term);
+    TEST(stopper(term));
+    TEST(stopper('Z' + stemmer(term)));
+
+    Xapian::StemStopper stopper_all_z(stemmer, Xapian::StemStopper::STEM_ALL_Z);
+    Xapian::StemStopper stopper_all(stemmer, Xapian::StemStopper::STEM_ALL);
+    term = "because";
+    stopper_all.add(term);
+    stopper_all_z.add(term);
+    TEST(!stopper_all_z(term));
+    TEST(stopper_all_z('Z' + stemmer(term)));
+    TEST(!stopper_all(term));
+    TEST(stopper_all('Z' + stemmer(term)));
+
+    Xapian::StemStopper stopper_none(stemmer, Xapian::StemStopper::STEM_NONE);
+    term = "and";
+    stopper_none.add(term);
+    TEST(stopper_none(term));
+    TEST(!stopper_none('Z' + stemmer(term)));
+
+    term = "cluster";
+    TEST(!stopper(term));
+    TEST(!stopper('Z' + stemmer(term)));
```

Comment:
A useful test, but it would be clearer to group this with the other `STEM_SOME` cases at the start.